pkg/cableengine/healthchecker/pinger.go

```diff
@@ -111,8 +114,8 @@ func (p *pingerInfo) Start() {
 			default:
 				err := p.doPing()
 				if err != nil {
-					klog.Errorf("Unable to start pinger for IP %q: %v", p.ip, err)
-					return
+					klog.Errorf("Unable to start pinger for IP %q: %v, retry in %d seconds", p.ip, err, doPingRetryTimeout)
+					time.Sleep(doPingRetryTimeout * time.Second)
```

Comment:
Coming late, but shouldnt it die after a few attempts?

This way a rogue pod can run forever if it constantly fails, which is quite a different behavior that what we had before (unless I'm missing something)

Reply:
The health-checker (aka pinger) is performed from the submariner-gateway pod itself and its objective is to run forever. Even during a success scenario pinger runs at regular interval and reports the RTT which can be seen in `subctl show ...` command.

Reply:
It runs as go func ( ~ separate thread) as part of GW pod , so k8s scheduler won't help us here if we'll exit this function sm-gw pod will continue running.

We'll be stuck without pinger running and no component will restart it (I think that was the bug root cause). 


Another direction could be to expose the pinger health somehow and let k8s scheduler monitor and restart the pod in case of failure (using liveness check)

Reply:
OK thanks this is clear now